### PR TITLE
1. Update demo.py and cases in demo 2. Add OpenXlab link in readme 3. update inference

### PR DESCRIPTION
**Pull request by ymzhang0319**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-from huggingface_hub import snapshot_download
-from moviepy.editor import AudioFileClip, VideoFileClip
-from transformers import CLIPImageProcessor, CLIPVisionModelWithProjection
 
-from diffusers import DDIMScheduler, EulerDiscreteScheduler, PNDMScheduler
-from foleycrafter.models.onset import torch_utils
-from foleycrafter.models.time_detector.model import VideoOnsetNet
-from foleycrafter.pipelines.auffusion_pipeline import Generator, denormalize_spectrogram
 from foleycrafter.utils.util import build_foleycrafter, read_frames_with_moviepy
+from foleycrafter.pipelines.auffusion_pipeline import denormalize_spectrogram
```
Comment (zengyh1900) [inline app.py:13]: have you ever use pre-commit to check the lint issue?

File: app.py [hunk 2]
```diff
@@ -40,16 +42,17 @@
 parser = ArgumentParser()
 parser.add_argument("--config", type=str, default="example/config/base.yaml")
 parser.add_argument("--server-name", type=str, default="0.0.0.0")
-parser.add_argument("--port", type=int, default=11451)
-parser.add_argument("--share", action="store_true")
+parser.add_argument("--port", type=int, default=7861)
+parser.add_argument("--share", type=bool, default=True)
```
Comment (zengyh1900) [inline app.py:46]: I think the default value shouldn't be True. Most users may run it locally.

File: app.py [hunk 3]
```diff
@@ -73,53 +74,47 @@ def load_model(self):
         print("Start Load Models...")
 
         # download ckpt
-        pretrained_model_name_or_path = "auffusion/auffusion-full-no-adapter"
+        pretrained_model_name_or_path = 'auffusion/auffusion-full-no-adapter'
         if not os.path.isdir(pretrained_model_name_or_path):
-            pretrained_model_name_or_path = snapshot_download(
-                pretrained_model_name_or_path, local_dir="models/auffusion"
-            )
+            pretrained_model_name_or_path = snapshot_download(pretrained_model_name_or_path, local_dir='models/auffusion')
```
Comment (zengyh1900) [inline app.py:79]: for `local_dir` which is used to save checkpoints, using `checkpoints` instead of `models` would be better.

File: app.py [hunk 4]
```diff
@@ -73,53 +74,47 @@ def load_model(self):
...
-        pretrained_model_name_or_path = "auffusion/auffusion-full-no-adapter"
+        pretrained_model_name_or_path = 'auffusion/auffusion-full-no-adapter'
         if not os.path.isdir(pretrained_model_name_or_path):
-            pretrained_model_name_or_path = snapshot_download(
-                pretrained_model_name_or_path, local_dir="models/auffusion"
-            )
+            pretrained_model_name_or_path = snapshot_download(pretrained_model_name_or_path, local_dir='models/auffusion') 
 
-        fc_ckpt = "ymzhang319/FoleyCrafter"
+        fc_ckpt = 'ymzhang319/FoleyCrafter'
```
Comment (zengyh1900) [inline app.py:81]: personal info?

File: app.py [hunk 5]
```diff
@@ -73,53 +74,47 @@ def load_model(self):
...
         if not os.path.isdir(fc_ckpt):
-            fc_ckpt = snapshot_download(fc_ckpt, local_dir="models/")
+            fc_ckpt = snapshot_download(fc_ckpt, local_dir='models/')
```
Comment (zengyh1900) [inline app.py:83]: check the directory path

File: assets/input/case1.mp4
Comment (zengyh1900) [inline assets/input/case1.mp4:1]: put these examples under `examples`

File: inference.py [hunk 6]
```diff
@@ -31,8 +31,8 @@ def args_parse():
     config.add_argument("--nprompt", type=str, default="", help="negative prompt for audio generation")
     config.add_argument("--seed", type=int, default=42, help="ramdom seed")
     config.add_argument("--temporal_align", action="store_true", help="use temporal adapter or not")
-    config.add_argument("--temporal_scale", type=float, default=1.0, help="temporal align scale")
-    config.add_argument("--semantic_scale", type=float, default=1.0, help="visual content scale")
+    config.add_argument('--temporal_align', type=bool, default=True, help='use temporal adapter or not')
```
Comment (zengyh1900) [inline inference.py:34]: we should use two types of scale instead of a boolean type for temporal alignment

**Review by zengyh1900 — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (ymzhang0319) [reply app.py:13]: Sorry, i'll remember it.

File: app.py (hunk 4 above)
Comment (ymzhang0319) [reply app.py:81]: Huggingface repo name.

File: inference.py [hunk 7]
```diff
@@ -55,7 +54,7 @@ def build_models(config):
 
     fc_ckpt = "ymzhang319/FoleyCrafter"
     if not os.path.isdir(fc_ckpt):
-        fc_ckpt = snapshot_download(fc_ckpt, local_dir="models/")
+        fc_ckpt = snapshot_download(fc_ckpt, local_dir="checkpoints/")
```
Comment (zengyh1900) [inline inference.py:57]: I think you should use `args.ckpt` ?

File: inference.py [hunk 8]
```diff
@@ -87,7 +86,7 @@ def build_models(config):
 
     # load semantic adapter
     pipe.load_ip_adapter(
-        "./models/semantic", subfolder="", weight_name="semantic_adapter.bin", image_encoder_folder=None
+        "./checkpoints/semantic", subfolder="", weight_name="semantic_adapter.bin", image_encoder_folder=None
```
Comment (zengyh1900) [inline inference.py:89]: `args.ckpt`?

**Review by zengyh1900 — CHANGES_REQUESTED** [review]

File: app.py [hunk 9]
```diff
@@ -55,13 +55,11 @@ class FoleyController:
     def __init__(self):
         # config dirs
         self.basedir = os.getcwd()
-        self.model_dir = os.path.join(self.basedir, "models")
+        self.model_dir = os.path.join(self.basedir, "checkpoints")
```
Comment (zengyh1900) [inline app.py:58]: I think it would be better to add another argument in `parser` for `checkpoints` instead of hard code here.

**Review by zengyh1900 — CHANGES_REQUESTED** [review]

**Review by zengyh1900 — APPROVED** [review]
lgtm

**Pull request merged**